### PR TITLE
Enhancement futures helpers module

### PR DESCRIPTION
**Pull request by dgou**
This is very similar to a module I have from "the day job" and thought it would be nice to share and jgt_common felt like it might be the right place to do so.

### CODE REVIEW

File: jgt_common/futures.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+def set_thread_pool_size(max_workers):
+    """Set the size for the shared ThreadPoolExecutor."""
+
+    global _MAX_WORKERS
+    _MAX_WORKERS = max_workers
+
+
+# Implemenation note:
+# This function is _not_ memoized:
+# If no executor is ever created, the shutdown function need do nothing.
```
[the remark this answers is not in the document]
Comment (dgou) [reply jgt_common/futures.py]: WF

**Comment by dgou** [comment]
@rbcasperson - understood.  concurrent.futures is one of those batteries included modules that I found about via @bradsbrown use of it in some behave cleanup code.
It baffled me for a long time because I thought it must be more complicated that it was.
The internal implementation is somewhat tricky "because threads", but the using futures is pretty straightforward. I was on the fence re: how much I wanted to add to the docs in terms of a primer for futures and why these functions would be helpful, and initially I fell off that fence on the assume-you-know-futures side... (at least for now)

**Comment by arounkles** [comment]
I like this. Simple, straightforward helper tooling. Add an example or two in documentation and you have a recipe for awesome.

File: jgt_common/futures.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+   configuration data to any function(s) that want to use these functions.
+
+   The executor thread pool defined here is mainly meant for use by the other
+   functions defined here, but users of this module are free to use it themselves
+   directly as needed.
+
+"""
+
+from concurrent.futures import ThreadPoolExecutor as _ThreadPoolExecutor
+from concurrent.futures import as_completed  # noqa - imported for pass-through use.
```
Comment (brolewis) [inline jgt_common/futures.py:35]: Is the `#noqa` still needed here since it's called multiple times below?

Comment (dgou) [reply jgt_common/futures.py:35]: I suppose not?  It's not imported with an underscore prefix for pass-through-ness, but that could be a plain old (no noqa) comment. WF.

Comment (brolewis) [reply jgt_common/futures.py:35]: Ah, I see what you mean. Yeah, I think the comment is fine but no need for the `noqa`

**Comment by dgou** [comment]
@brolewis New commit pushed. I guess github doesn't slack-spam on that action :-)

**Pull request merged**